libbeat/processors/decompress/gzip.go

```diff
@@ -0,0 +1,147 @@
+		FailOnError: true,
+	}
+)
+
+func init() {
+	processors.RegisterPlugin("decompress_gzip_fields",
+		checks.ConfigChecked(NewDecompressGzipFields,
+			checks.RequireFields("fields"),
+			checks.AllowedFields("fields", "ignore_missing", "overwrite_keys", "overwrite_keys", "fail_on_error", "when")))
+}
+
+// NewDecompressGzipFields construct a new decompress_gzip_fields processor.
+func NewDecompressGzipFields(c *common.Config) (processors.Processor, error) {
+	config := defaultDecompressGzipFieldsConfig
+
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unpack the decompress_gzip_fields configuration: %s", err)
+	}
+	if len(config.Fields) == 0 {
+		return nil, errors.New("no fields to decompress configured")
+	}
+	f := &decompressGzipFields{decompressGzipFieldsConfig: config}
+
+	// Set fields as string -> string
+	f.fields = make(map[string]string, len(config.Fields))
+	for src, dstIf := range config.Fields.Flatten() {
+		dst, ok := dstIf.(string)
+		if !ok {
+			return nil, errors.Errorf("bad destination mapping for %s: destination field must be string, not %T (got %v)", src, dstIf, dstIf)
+		}
+		f.fields[src] = dst
+	}
+	return f, nil
+}
+
+// Run applies the decompress_gzip_fields processor to an event.
+func (f *decompressGzipFields) Run(event *beat.Event) (*beat.Event, error) {
+	saved := *event
+	if f.FailOnError {
+		saved.Fields = event.Fields.Clone()
+		saved.Meta = event.Meta.Clone()
+	}
+	for src, dest := range f.fields {
+		if err := f.decompressGzipField(src, dest, event); err != nil && f.FailOnError {
+			return &saved, err
+		}
+	}
+	return event, nil
+}
+
+func (f *decompressGzipFields) decompressGzipField(src, dest string, event *beat.Event) error {
+	// Check source value
+	data, err := event.GetValue(src)
+	if err != nil {
+		if f.IgnoreMissing && errors.Cause(err) == common.ErrKeyNotFound {
+			return nil
+		}
+		return errors.Wrapf(err, "could not fetch value for field %s", src)
+	}
+
+	text, ok := data.(string)
+	if !ok {
+		return errors.Errorf("field %s is not of string type", src)
+	}
+
+	//Uncompress gzip message
+	b := bytes.NewBuffer([]byte(text))
+
+	var r io.Reader
+	r, err = gzip.NewReader(b)
+
+	if err != nil {
+		return errors.Wrapf(err, "error decompressing field %s", src)
+	}
+
+	var resB bytes.Buffer
+	resB.ReadFrom(r)
+
+	decompressed := string(resB.Bytes())
+
+	// Write message out
+	if src != dest && !f.OverwriteKeys {
+		if _, err = event.GetValue(dest); err == nil {
+			return errors.Errorf("target field %s already has a value. Set the overwrite_keys flag or drop/rename the field first", dest)
+		}
+	}
+	if _, err = event.PutValue(dest, decompressed); err != nil {
+		return errors.Wrapf(err, "failed setting field %s", dest)
+	}
+	return nil
+}
+
+// String returns a string representation of this processor.
+func (f decompressGzipFields) String() string {
+	json, _ := json.Marshal(f.decompressGzipFieldsConfig)
+	return "decompress_gzip_fields=" + string(json)
```

Comment:
I think displaying the name of the fields which are decompressed is enough.

Reply:
I have made a change to what i *think* you mean